source/Stargate-Examples-Tests/SouthAmericanCurrenciesRESTfulControllerTest.class.st

```diff
@@ -25,15 +25,41 @@ SouthAmericanCurrenciesRESTfulControllerTest >> setUpResourceController [
 	resourceController := SouthAmericanCurrenciesRESTfulController new
 ]
 
+{ #category : #tests }
+SouthAmericanCurrenciesRESTfulControllerTest >> testCantGetArgentinePesoAskingForAnEmptyLanguage [
+
+	| request |
+
+	request := self
+	           requestToGETResourceIdentifiedBy: 'ARS'
+	           representedAs:
+	           resourceController currencyVersion1dot0dot0MediaType
+	           in: ''.
+
+	self
+		should: [ resourceController currencyBasedOn: request within: self newHttpRequestContext ]
+		raise: HTTPNotAcceptable
+		withExceptionDo: [ :error | 
+			self
+				assert: error messageText
+					equals: 'Cannot generate a response entity with acceptable characteristics.';
+				assert: error code equals: 406;
+				assert: error allowedLanguageTags size equals: 2;
+				assert: error allowedLanguageTags includes: 'en-US' asLanguageTag;
+				assert: error allowedLanguageTags includes: 'es-AR' asLanguageTag
+			]
+]
+
 { #category : #tests }
 SouthAmericanCurrenciesRESTfulControllerTest >> testCantGetArgentinePesoAskingForUnsupportedLanguage [
 
 	| request |
 
 	request := self
-		requestToGETResourceIdentifiedBy: 'ARS'
-		accepting: resourceController currencyVersion1dot0dot0MediaType.
-	request setAcceptLanguage: 'de'.
+	           requestToGETResourceIdentifiedBy: 'ARS'
+	           representedAs:
+	           resourceController currencyVersion1dot0dot0MediaType
+	           in: 'de'.
```

Comment:
I would perform a set of extract methods for the language tags.

'de' -> unsupportedLanguage
'en' -> english
'es' -> spanish
'es-AR' -> spanishAsUsedInArgentina or argentineSpanish

You can learn more here https://www.w3.org/International/articles/language-tags/